index.html

```diff
@@ -12,451 +12,451 @@
+                      class="min-temp-container m-0 rounded-2xl px-2 py-1.5 h-fit"
                       id="min-temp-container"
                     ></div>
                     <div
-                      class="max-temp-container"
+                      class="max-temp-container m-0 rounded-2xl px-2 py-1.5 h-fit"
                       id="max-temp-container"
                     ></div>
                   </div>
                 </div>
 
-                <div class="daily-forecast-container" id="forecast-day-7">
-                  <div class="forecast-item-info" id="forecast-date"></div>
-                  <div class="forecast-item-info" id="forecast-weather"></div>
-                  <div class="forecast-item-info" id="forecast-temp">
+                <div class="daily-forecast-container h-full grid justify-evenly items-center mx-2 my-0  bg-forecastColor rounded-md text-center overflow-hidden first:ml-0"  id="forecast-day-7">
+                  <div class="forecast-item-info text-white text-xl m-0 p-0" id="forecast-date"></div>
+                  <div class="forecast-item-info font-semibold text-6xl justify-self-center self-center" id="forecast-weather"></div>
+                  <div class="forecast-item-info text-white text-base m-0 p-0 flex flex-col justify-center gap-1.5 h-full" id="forecast-temp">
                     <div
-                      class="min-temp-container"
+                      class="min-temp-container m-0 rounded-2xl px-2 py-1.5 h-fit"
                       id="min-temp-container"
                     ></div>
                     <div
-                      class="max-temp-container"
+                      class="max-temp-container m-0 rounded-2xl px-2 py-1.5 h-fit"
                       id="max-temp-container"
                     ></div>
                   </div>
                 </div>
               </div>
 
-              <div id="johannesburg-container" class="city-forecast">
-                <div class="daily-forecast-container" id="forecast-day-1">
-                  <div class="forecast-item-info" id="forecast-date"></div>
-                  <div class="forecast-item-info" id="forecast-weather"></div>
-                  <div class="forecast-item-info" id="forecast-temp">
+              <div id="johannesburg-container" class="city-forecast w-full h-full flex flex-row justify-between content-center overflow-x-auto">
+                <div class="daily-forecast-container h-full grid justify-evenly items-center mx-2 my-0  bg-forecastColor rounded-md text-center overflow-hidden first:ml-0" id="forecast-day-1">
+                  <div class="forecast-item-info text-white text-xl m-0 p-0" id="forecast-date"></div>
+                  <div class="forecast-item-info font-semibold text-6xl justify-self-center self-center" id="forecast-weather"></div>
+                  <div class="forecast-item-info text-white text-base m-0 p-0 flex flex-col justify-center gap-1.5 h-full" id="forecast-temp">
```

Comment:
ID's generally should not repeat, if you want repeated styles you should rather use classNames.

Reply:
I use these IDs repeatedly, because I am using the ids to style the elements as each of the three elements have the same class for DOM manipulation purposes, but different styling and these IDs re-occur for the exact same elements for each forecast-item.